bundles/cif-connector-graphql/src/main/java/com/adobe/cq/commerce/gui/components/configuration/servlets/ConfigurationInheritanceServlet.java

```diff
@@ -0,0 +1,138 @@
+import org.apache.sling.api.resource.ValueMap;
+import org.apache.sling.api.servlets.SlingSafeMethodsServlet;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.adobe.cq.commerce.virtual.catalog.data.Constants;
+
+@Component(
+    service = Servlet.class,
+    name = "ConfigurationInheritanceServlet",
+    immediate = true,
+    property = {
+        "sling.servlet.methods=GET",
+        "sling.servlet.selectors=cifconfig",
+        "sling.servlet.extensions=json",
+        "sling.servlet.resourceTypes=sling/servlet/default"
+    })
+public class ConfigurationInheritanceServlet extends SlingSafeMethodsServlet {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ConfigurationInheritanceServlet.class);
+    private final static String[] CONFIG_KEYS = {
+        "cq:catalogDataResourceProviderFactory",
+        "cq:catalogIdentifier",
+        "cq:graphqlClient",
+        "jcr:language",
+        "magentoGraphqlEndpoint",
+        "magentoRootCategoryId",
+        "magentoStore"
+    };
+
+    protected void doGet(SlingHttpServletRequest request, SlingHttpServletResponse response) throws IOException {
+
+        try {
+            Resource resource = request.getResource();
+            String resourcePath = resource.getPath();
+
+            // Get the container resource path
+            String configContainerPath = resourcePath.endsWith(Constants.COMMERCE_BUCKET_PATH) ? resourcePath.replace(
+                "/" + Constants.COMMERCE_BUCKET_PATH, "") : resourcePath;
+
+            // Get the container resource
+            Resource configContainerResource = request.getResourceResolver().getResource(configContainerPath);
+
+            if (configContainerResource == null) {
+                response.sendError(500, "Unable to load configuration container resource");
+                return;
+            }
+
+            // Start with a JSON documents that assumes no inheritance
+            JSONObject config = new JSONObject();
+            config.put("inherited", false);
+
+            // Get the parent resource to start gathering inherited properties
+            Resource parentConfigContainerResource = configContainerResource.getParent();
+
+            // If the resource is still under /conf
+            if (configContainerPath.startsWith("/conf") &&
+                parentConfigContainerResource != null &&
+                !parentConfigContainerResource.getPath().equals("/conf")) {
+
+                // Set inherited prop true and add a new JSON object to hold inherited properties values
+                config.put("inherited", true);
+                JSONObject inheritedProps = new JSONObject();
+
+                Resource parentConfig;
+                while (parentConfigContainerResource != null && !parentConfigContainerResource.getPath().equals("/conf")) {
```

Comment:
Here you're checking for inheritance within `/conf`. What about `/libs/settings/cloudconfigs/commerce` and a potential overlay in `/apps`?